icu4c/source/common/locid.cpp

```diff
@@ -424,6 +424,12 @@ Locale::Locale(const Locale &other)
     *this = other;
 }
 
+Locale::Locale(Locale&& other) U_NOEXCEPT
+    : UObject(other), fullName(fullNameBuffer), baseName(fullName) {
+  // Not using std::move() for that would require STL header files.
```

Comment:
std::move() should be fine, I think. Discuss?
In particular, we should not have to contort ourselves for using move semantics.

We can't use most of the STL where that might throw exceptions (ICU neither throws nor handles them) or allocate memory (ICU memory allocation is overridable and does not throw exceptions). So no string, no vector, no dynamic_cast(reference), but yes dynamic_cast(pointer), ... I think we also use C++11 atomics & mutexes, at least in some configurations.

Reply:
You probably need to null the pointers after freeing them and before moving from other, to make the second line true here.

Reply:
I've definitely used `std::move` in ICU implementation code, and it's been fine.

I don't see what markus asked about freeing pointers in this function.  This is a constructor so there is nothing to free yet.